File: Selenium/Project/PrActivity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up Firefox driver
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    # Navigate to the OrangeHRM login page
    driver.get("http://alchemy.hguy.co/orangehrm")

    # Find and enter the username
    driver.find_element(By.ID, "txtUsername").send_keys("orange")
    # Find and enter the password
    driver.find_element(By.ID, "txtPassword").send_keys("orangepassword123")
    # Click on the login button
    driver.find_element(By.ID, "btnLogin").click()

    # Navigate to the My Info page
    driver.find_element(By.ID, "menu_pim_viewMyDetails").click()

    # Click on the Emergency Contacts menu item
    driver.find_element(By.LINK_TEXT, "Emergency Contacts").click()

    # Retrieve information about all the contacts listed in the table
    contact_rows = driver.find_elements(By.XPATH, "//table[@id='emgcontact_list']//tr")[1:]  # Exclude header row
    for row in contact_rows:
        cells = row.find_elements(By.TAG_NAME, "td")
        contact_name = cells[0].text
        relationship = cells[1].text
        home_telephone = cells[2].text
        mobile_telephone = cells[3].text
        work_telephone = cells[4].text
        print(f"Contact Name: {contact_name}, Relationship: {relationship}, Home Phone: {home_telephone}, "
              f"Mobile Phone: {mobile_telephone}, Work Phone: {work_telephone}")

# Close the browser
driver.quit()

File: Selenium/Project/PrActivity9.java
package Activities;


import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.testng.annotations.*;
        import java.util.List;

public class PrActivity9 extends BaseClass {

    @Test
    public void retrieveEmergencyContacts() {
        setUp();
        login();

        // Click on My Info menu
        driver.findElement(By.id("menu_pim_viewMyDetails")).click();

        // Click on the "Emergency Contacts" menu item
        driver.findElement(By.linkText("Emergency Contacts")).click();

        // Retrieve information about all the contacts listed in the table
        List<WebElement> contactRows = driver.findElements(By.xpath("//table[@id='emgcontact_list']/tbody/tr"));

        System.out.println("Emergency Contacts:");

        for (WebElement row : contactRows) {
            List<WebElement> columns = row.findElements(By.tagName("td"));
            String name = columns.get(1).getText();
            String relationship = columns.get(2).getText();
            String homeTelephone = columns.get(3).getText();
            String mobileTelephone = columns.get(4).getText();

            System.out.println("Name: " + name);
            System.out.println("Relationship: " + relationship);
            System.out.println("Home Telephone: " + homeTelephone);
            System.out.println("Mobile Telephone: " + mobileTelephone);
            System.out.println();
        }

        // Close the browser after all contacts are processed
        tearDown();
    }
}